idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2813 @@
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Optimal Solution Found.\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "{'Problem': [{'Lower bound': -inf, 'Upper bound': inf, 'Number of objectives': 1, 'Number of constraints': 16, 'Number of variables': 16, 'Sense': 'unknown'}], 'Solver': [{'Status': 'ok', 'Message': 'Ipopt 3.13.2\\\\x3a Optimal Solution Found', 'Termination condition': 'optimal', 'Id': 0, 'Error rc': 0, 'Time': 0.0767662525177002}], 'Solution': [OrderedDict([('number of solutions', 0), ('number of solutions displayed', 0)])]}"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "solver.solve(m, tee=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b27b6b8",
+   "metadata": {},
+   "source": [
+    "This is a good sign that the model solved optimally and a solution was found. Just to be sure we would check if there are any more numerical issues by calling `report_numerical_issues()`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "988dc8fd",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "====================================================================================\n",
+      "Model Statistics\n",
+      "\n",
+      "    Jacobian Condition Number: 7.955E+03\n",
+      "\n",
+      "------------------------------------------------------------------------------------\n",
+      "1 WARNINGS\n",
+      "\n",
+      "    WARNING: 3 Variables at or outside bounds (tol=0.0E+00)\n",
+      "\n",
+      "------------------------------------------------------------------------------------\n",
+      "2 Cautions\n",
+      "\n",
+      "    Caution: 3 Variables with value close to their bounds (abs=1.0E-04, rel=1.0E-04)\n",
+      "    Caution: 5 Variables with value close to zero (tol=1.0E-08)\n",
+      "\n",
+      "------------------------------------------------------------------------------------\n",
+      "Suggested next steps:\n",
+      "\n",
+      "    display_variables_at_or_outside_bounds()\n",
+      "\n",
+      "====================================================================================\n"
+     ]
+    }
+   ],
+   "source": [
+    "dt.report_numerical_issues()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "75e9d9b8",
+   "metadata": {},
+   "source": [
+    "Here there are some numerical issues with the variables at or outside the bounds which is a physical condition of being a pure stream and thus the salt concentration would be 0. Thus baring this there are no more issues, and thus we can say that the model has been debugged. A detailed notebook on using `DiagnosticsToolbox` can be found [here](../../diagnostics/degeneracy_hunter.ipynb).\n",
```

Comment:
This is a nice reference, move this to the first description at the start of the diagnostics section.